### DISCUSSION TITLE
Is there any way resolve YesSql.ISession dependency inside a static class?

### DISCUSSION THREAD

**Discussion by AmadeuAntunes** [OrchardCMS/OrchardCore#8450] [Q&A]
As it is a static class I can't use contructor to get di of YesSql.ISession

Is there any static class that I could call to get that instance?

**Comment by hishamco** [comment]
This is a common scenario in ASP.NET Core, but simply you can create a method called `Initialize()` that accepts `HttpContext` from startup configuration to be used in your `static` class, then you can looking for your required services

**Reply by AmadeuAntunes** [reply]
I've try something like Extensions.Configure(app.ApplicationServices.GetService<YesSql.ISession>()); but it doesnt worked

**Comment by deanmarcussen** [comment]
I think the issue here might be the concept of the static class. 

an `ISession` is scoped to a request, so you can't just resolve it once, and continue to work with it during a static classes lifecyle, i.e. forever.

Perhaps if you share the explanation of why you need a static class, it might help. I am wondering if they are extension methods, because you have used the `Extensions` word, but need a bit more information really.

**Reply by AmadeuAntunes** [reply]
I needed create an **extension** for `public static string GetVersion(this IContent contentItem, YesSql.ISession session)`
`

**Reply by deanmarcussen** [reply]
I would suggest following this technique using the `IOrchardHelper`, which can be injected where you need it.

https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore/OrchardCore.ContentLocalization.Abstractions/ContentLocalizationOrchardHelperExtensions.cs

**Reply by jtkech** [reply]
If you are always executing in a scope context, just for the fun try

    using OrchardCore.Environment.Shell.Scope

        var session = ShellScope.Services.GetRequiredService<ISession>();

But what @deanmarcussen suggested is a more recommended way ;)
